python/lsst/pipe/tasks/multiBand.py

```diff
@@ -532,8 +539,8 @@ def readSources(self, dataRef):
         merged = dataRef.get(self.config.coaddName + "Coadd_mergeDet", immediate=True)
         self.log.info("Read %d detections: %s", len(merged), dataRef.dataId)
         idFactory = self.makeIdFactory(dataRef)
-        for s in merged:
-            idFactory.notify(s.getId())
+        maxId = np.max(merged["id"])
```

Comment:
I have no idea what's going on here, so a inline comments would help a hapless reader like me know what's special about the max id.

Reply:
I'll copypasta the comment from: https://github.com/lsst/pipe_tasks/blob/main/python/lsst/pipe/tasks/deblendCoaddSourcesPipeline.py#L158-L163
(I was just updating here to match what's there which gets rid of the loop).